### DISCUSSION TITLE
Is it possible to change the text animations to discrete character appearance?

### DISCUSSION THREAD

**Discussion by MartinXPN** [ManimCommunity/manim#3008] [Q&A] [answered]
Hey manim team!
I'm quite new to the manim library, and I'd like to create several tutorials on algorithms that include displaying code.
I was wondering if it's possible to change the text appearance animation to something that would animate each character's appearance discretely. An example: https://media.tenor.com/_DOBjnGspYAAAAAM/code-coding.gif

![](https://media.tenor.com/_DOBjnGspYAAAAAM/code-coding.gif)

Any hints/directions would be really appreciated!

**Comment by MrDiver** [comment, marked as answer]
So you have some different options. The first thing you should take a look at is the animation docs, there you can find all the animations your heart desires.

The simplest solution you have is to use the [`AddTextLetterByLetter`](https://docs.manim.community/en/stable/reference/manim.animation.creation.AddTextLetterByLetter.html) Animation.

Example:

```py
class TextAppearances(Scene):
    def construct(self):
        text = Text("Hello World").scale(2)
        self.play(AddTextLetterByLetter(text))
        self.wait()
```
![TextAppearances_ManimCE_v0 16 0 post0](https://user-images.githubusercontent.com/17803932/200201220-2c5f902a-c0ae-4535-8dd3-4e0ea3a464d2.gif)

Otherwise you can just make it more custom because in the end its just python the end like this for example:


```py
class TextAppearances2(Scene):
    def construct(self):
        text = Text("Hello World").scale(2)
        group = AnimationGroup(
            *[FadeIn(char, shift=DOWN) for char in text], lag_ratio=0.5, run_time=2
        )
        self.play(group)
        self.wait()
```
![TextAppearances2_ManimCE_v0 16 0 post0](https://user-images.githubusercontent.com/17803932/200201269-92601287-f1ab-42bc-ae7a-5d5eb600e198.gif)

I hope this helps you a little bit, have fun animating 🥳!

Links for Animations: https://docs.manim.community/en/stable/reference_index/animations.html

**Comment by MrDiver** [comment]
And experimenting around you can just write a custom animation function

```py
def PlayTerminalWrite(scene, text, run_time, cursor_size=0.5):
    snippet_time = run_time / len(text)
    cursor = Rectangle(
        height=cursor_size, width=cursor_size * 0.5, color=WHITE, fill_opacity=1
    )
    cursor.move_to(text[0].get_center())
    scene.play(Write(cursor))
    for char in text:
        scene.play(
            AnimationGroup(
                cursor.animate.next_to(
                    [char.get_x(), cursor.get_y(), 0],
                    RIGHT,
                    buff=0.5,
                ),
                FadeIn(char),
                run_time=snippet_time,
                lag_ratio=0.5,
            )
        )
    scene.play(FadeOut(cursor))


class TextAppearances3(Scene):
    def construct(self):
        text = Text("Writing like a Terminal").shift(UP)
        text2 = Text("But Animated").shift(DOWN)
        PlayTerminalWrite(self, text, 1)
        self.wait()
        PlayTerminalWrite(self, text2, 1)
        self.wait()
```
![TextAppearances3_ManimCE_v0 16 0 post0](https://user-images.githubusercontent.com/17803932/200202794-53a173d6-738c-4733-883f-7cce3aba5887.gif)

**Reply by MartinXPN** [reply]
This looks amazing! Thank you so much!
